source/common/http/http1/codec_impl.cc

```diff
@@ -501,8 +510,22 @@ Status ConnectionImpl::completeLastHeader() {
     // in ConnectionImpl::onHeaderValue. http_parser does not strip leading or trailing whitespace
     // as the spec requires: https://tools.ietf.org/html/rfc7230#section-3.2.4
     current_header_value_.rtrim();
+
     headers_or_trailers.addViaMove(std::move(current_header_field_),
                                    std::move(current_header_value_));
+
+    if (codec_settings_.header_key_format_ == Http1Settings::HeaderKeyFormat::PreservedCase) {
```

Comment:
Sorry for duplicating comments, I think this should call `headers_or_trailers.addCasePreserved` or something instead of altering the map